jwst/ami/matrix_dft.py

```diff
@@ -0,0 +1,352 @@
+        Vs = (np.arange(npixY) - npixY / 2) * dV
+    elif centering == ADJUSTABLE:
+        if offset is None:
+            offsetY, offsetX = 0.0, 0.0
+        else:
+            try:
+                offsetY, offsetX = offset
+            except ValueError:
+                raise ValueError(
+                    "'offset' must be supplied as a 2-tuple with "
+                    "(y_offset, x_offset) as floating point values"
+                )
+        Xs = (np.arange(npupX) - float(npupX) / 2.0 - offsetX + 0.5) * dX
+        Ys = (np.arange(npupY) - float(npupY) / 2.0 - offsetY + 0.5) * dY
+
+        Us = (np.arange(npixX) - float(npixX) / 2.0 - offsetX + 0.5) * dU
+        Vs = (np.arange(npixY) - float(npixY) / 2.0 - offsetY + 0.5) * dV
+    elif centering == SYMMETRIC:
+        Xs = (np.arange(npupX) - float(npupX) / 2.0 + 0.5) * dX
+        Ys = (np.arange(npupY) - float(npupY) / 2.0 + 0.5) * dY
+
+        Us = (np.arange(npixX) - float(npixX) / 2.0 + 0.5) * dU
+        Vs = (np.arange(npixY) - float(npixY) / 2.0 + 0.5) * dV
+    else:
+        raise ValueError("Invalid centering style")
+
+    XU = np.outer(Xs, Us)
+    YV = np.outer(Ys, Vs)
+
+    if inverse:
+        expYV = np.exp(-2.0 * np.pi * -1j * YV).T
+        expXU = np.exp(-2.0 * np.pi * -1j * XU)
+        t1 = np.dot(expYV, plane)
+        t2 = np.dot(t1, expXU)
+    else:
+        expXU = np.exp(-2.0 * np.pi * 1j * XU)
+        expYV = np.exp(-2.0 * np.pi * 1j * YV).T
+        t1 = np.dot(expYV, plane)
+        t2 = np.dot(t1, expXU)
+
+    norm_coeff = np.sqrt((nlamDY * nlamDX) / (npupY * npupX * npixY * npixX))
+
+    return norm_coeff * t2
+
+
+def matrix_idft(*args, **kwargs):
+    kwargs['inverse'] = True
+    return matrix_dft(*args, **kwargs)
+
+
+matrix_idft.__doc__ = matrix_dft.__doc__.replace(
+    'Perform a matrix discrete Fourier transform',
+    'Perform an inverse matrix discrete Fourier transform'
+)
+
+
+class MatrixFourierTransform:
+    """Implements a discrete matrix Fourier transform for optical propagation,
+    following the algorithms discussed in Soummer et al. 2007 JOSA 15 24.
+
+    Parameters
+    ----------
+    centering : {'FFTSTYLE', 'SYMMETRIC', 'ADJUSTABLE'}, optional
+        What type of centering convention should be used for this FFT?
+        * ADJUSTABLE (the default) For an output array with ODD size n,
+          the PSF center will be at the center of pixel (n-1)/2. For an output
+          array with EVEN size n, the PSF center will be in the corner between
+          pixel (n/2-1, n/2-1) and (n/2, n/2)
+        * FFTSTYLE puts the zero-order term in a single pixel.
+        * SYMMETRIC spreads the zero-order term evenly between the center
+          four pixels
+    verbose : bool
+        Deprecated. Use poppy.conf.default_logging_level to set DEBUG level
+        logging.
+
+    Example
+    -------
+    >>> mft = MatrixFourierTransform()
+    >>> result = mft.perform(pupilArray, focalplane_size, focalplane_npix)
```

Comment:
This line appears to be killing all of the CI tests, due to a problem with building the doc string. The argument names don't match what's used in the actual function definition.